core/src/test/scala/org/apache/spark/storage/BlockManagerDecommissionIntegrationSuite.scala

```diff
@@ -87,36 +113,30 @@ class BlockManagerDecommissionIntegrationSuite extends SparkFunSuite with LocalS
     }
 
     // Listen for the job & block updates
-    val taskStartSem = new Semaphore(0)
-    val broadcastSem = new Semaphore(0)
     val executorRemovedSem = new Semaphore(0)
-    val taskEndEvents = ArrayBuffer.empty[SparkListenerTaskEnd]
+    val taskEndEvents = new ConcurrentLinkedQueue[SparkListenerTaskEnd]()
     val blocksUpdated = ArrayBuffer.empty[SparkListenerBlockUpdated]
-    sc.addSparkListener(new SparkListener {
 
+    def getCandidateExecutorToDecom: Option[String] = if (whenToDecom == TaskStarted) {
+      accum.value.asScala.headOption
```

Comment:
I don't think this is going to work as intended, accumulators send updates back at the end of the task, unless something has changed.

Reply:
If you don't believe that is the case, can we add an assertion in here that none of the tasks have finished yet?